pygly/Monosaccharide.py

```diff
@@ -1,1148 +1,1317 @@
+        # It may be None, representing unset
+        self._config = None
+
+        # Stem 3-letter code [glc, gal, man, rib, gro]
+        # It may be None, representing unset
+        self._stem = None
+
+        # Superclass (based on the number of consecutive carbon atoms)
+        self._superclass = None
+
+        # Ring closure starting position (THE NUMBER OF THE CARBON) (int)
+        self._ring_start = None
+            
+        # Ring closure ending position (THE NUMBER OF THE CARBON) (int)
+        self._ring_end = None
+
+        # Modifier type (Optional, Repeatable)
+        self._mods = []
+
+        # A list of 0 or more links to substituent objects.
+        self._substituent_links = []
+
+        # Short for external descriptor(eg. WURCS) identifier
+        # Could be used for keeping track of monosaccharide from descriptor assigned ID
+        self._eid = None
+
+    def is_monosaccharide(self):
+        return True
+
+    def clone(self):
+        m = Monosaccharide()
+        m._anomer = self._anomer
+        m._config = copy.deepcopy(self._config)
+        m._stem = copy.deepcopy(self._stem)
+        m._superclass = self._superclass
+        m._ring_start = self._ring_start
+        m._ring_end = self._ring_end
+        m._mods = copy.deepcopy(self._mods)
+        # m._composition = copy.copy(self._composition)
+        # Will this do it?
+        m._substituent_links = copy.deepcopy(self._substituent_links)
+        for l in m._substituent_links:
+            l.set_parent(m)
+            subst = l.child()
+            subst._links = []
+        # m._links = copy.deepcopy(self._links)
+        m._id = self._id
+        m._connected = self._connected
+        m._external_descriptor = self._external_descriptor
+        return m
+
+    def noring(self):
+        return (self.ring() == (0,0))
+
+    def deepclone(self,identified_link=None, cache=None, rep_bridge_undone=[]):
+
+        m = self.clone()
+        if cache == None:
+            cache = dict()
+            cache[m.id()] = m
+        identified_link_copy = None
+
+        for l in self._links:
+            assert l.child().id() != None
+
+            if l.repeat_bridge_link():
+                rep_bridge_undone.append(l)
+                continue
+
+            if l.child().id() not in cache:
+                if identified_link:
+                    c,idlc = l.child().deepclone(identified_link=identified_link,cache=cache,rep_bridge_undone=rep_bridge_undone)
+                else:
+                    c = l.child().deepclone(cache=cache,rep_bridge_undone=rep_bridge_undone)
+                    idlc = None
+                cache[l.child().id()] = c
+            else:
+                c = cache[l.child().id()]
+                idlc = None
+
+            cl = copy.deepcopy(l)
+            cl.set_child(c)
+            # TODO
+            if isinstance(l.parent(), Monosaccharide):
+                cl.set_parent(m)
+            else:
+                raise RuntimeError
```

Comment:
I got this RuntimeError when trying to do the GNOme computation, presumably because of a substituent as parent. Why is this condition necessary? What are you trying to detect/avoid? The point of the Node class with the parent/child logic is so that both Monosaccharide and Substituent can participate as parents/children.

Reply:
I removed this if statement (code back the way it was) which permits structures with substituent in links to be cloned. Why was this check added?